src/trio/_core/_tests/test_run.py

```diff
@@ -2044,9 +2044,7 @@ async def __anext__(self) -> list[int]:
 
             return items
 
-    result: list[list[int]] = []
-    async for vals in async_zip(it(4), it(2)):
-        result.append(vals)
+    result: list[list[int]] = [vals async for vals in async_zip(it(4), it(2))]
```

Comment:
Thinking about it, actually it'd be good to avoid comprehensions in test code, if they're the "main loop". If an error occurs, a comprehension would just be discarded while an explicit loop would leave the partial results behind to be printed out with other local variables.

Reply:
Reverted change in 17afaab